### PR TITLE
feat: Windows moved by keyboard now attach to nearest side of branch

### PR DESCRIPTION
**Pull request by mmstick**

### CODE REVIEW

**Review by jacobgkau — APPROVED** [review]
Regression testing passed:
[shell-pr815-groovy.txt](https://github.com/pop-os/shell/files/5788726/shell-pr815-groovy.txt)
[shell-pr815-focal.txt](https://github.com/pop-os/shell/files/5788788/shell-pr815-focal.txt)

It does look like this sometimes doesn't work when windows are oversized/overlapping (but it's still an improvement):
- Before: https://www.dropbox.com/s/hi57848wacpwnpq/shell-pr815-screencast1.mkv?dl=0 
- After (w/ oversized window at the end): https://www.dropbox.com/s/do6tkonxm57uq1h/shell-pr815-screencast2.mkv?dl=0 

I'm approving, but I can re-review if you'd like to try and address that case.

**Comment by mmstick** [comment]
I've an idea how I can fix it

**Review by jacobgkau — APPROVED** [review]
Movement is working perfectly now: https://www.dropbox.com/s/r3bp9glpjrxee2v/shell-pr815-screencast5.mkv?dl=0 

Regression testing still passing:
[shell-pr815-groovy-run2.txt](https://github.com/pop-os/shell/files/5803742/shell-pr815-groovy-run2.txt)
[shell-pr815-focal-run2.txt](https://github.com/pop-os/shell/files/5803832/shell-pr815-focal-run2.txt)

It does look like https://github.com/pop-os/shell/pull/821 is not in this branch, and I do not have an `Update Branch` button, but that should not be the case once this is merged.

shell-pr815-screenshot1

**Pull request merged**
